omod/src/main/java/org/openmrs/module/mksreports/web/controller/MKSReportsManageController.java

```diff
@@ -130,7 +145,13 @@ public void renderPatientHistory(ModelMap model, HttpServletRequest request, Htt
 			byte[] pdfBytes = outStream.toByteArray();
 			response.setContentLength(pdfBytes.length);
 			response.setContentType("application/pdf");
-			response.addHeader("Content-Disposition", "attachment;filename=patientHistory.pdf");
+			
+			// set the file as an attachment with the suggested filename if the GET params
+			// did not indicate it's being opened in another tab
+			if (StringUtils.isBlank(target)) {
+				response.addHeader("Content-Disposition", "attachment;filename=patientHistory.pdf");
```

Comment:
This might not be the patient history right?
I don't have a very good idea as to how to name this file, but I'm guessing it should probably be scrambled a little. Maybe "a UUID" dot "pdf"?